pkg/auth/handler/loginid/add.go

```diff
@@ -0,0 +1,171 @@
+package loginid
+
+import (
+	"net/http"
+
+	"github.com/skygeario/skygear-server/pkg/auth"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/hook"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/principal"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/principal/password"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/userprofile"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/userverify"
+	"github.com/skygeario/skygear-server/pkg/auth/event"
+	"github.com/skygeario/skygear-server/pkg/auth/model"
+	coreAuth "github.com/skygeario/skygear-server/pkg/core/auth"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authinfo"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz"
+	"github.com/skygeario/skygear-server/pkg/core/auth/authz/policy"
+	"github.com/skygeario/skygear-server/pkg/core/db"
+	"github.com/skygeario/skygear-server/pkg/core/handler"
+	"github.com/skygeario/skygear-server/pkg/core/inject"
+	"github.com/skygeario/skygear-server/pkg/core/server"
+	"github.com/skygeario/skygear-server/pkg/core/validation"
+)
+
+func AttachAddLoginIDHandler(
+	server *server.Server,
+	authDependency auth.DependencyMap,
+) *server.Server {
+	server.Handle("/login_id/add", &AddLoginIDHandlerFactory{
+		authDependency,
+	}).Methods("OPTIONS", "POST")
+	return server
+}
+
+type AddLoginIDHandlerFactory struct {
+	Dependency auth.DependencyMap
+}
+
+func (f AddLoginIDHandlerFactory) NewHandler(request *http.Request) http.Handler {
+	h := &AddLoginIDHandler{}
+	inject.DefaultRequestInject(h, f.Dependency, request)
+	return h.RequireAuthz(h, h)
+}
+
+type AddLoginIDRequestPayload struct {
+	password.LoginID
+}
+
+// @JSONSchema
+const AddLoginIDRequestSchema = `
+{
+	"$id": "#AddLoginIDRequest",
+	"type": "object",
+	"properties": {
+		"key": { "type": "string", "minLength": 1 },
```

Comment:
Should we use `login_id_key` and `login_id` for consistency?

Reply:
This shape (`key` and `value` for login IDs) is previously used for signup API, so I don't think we need to change it.

`login_id_key` and `login_id` is used in login API, since using `key` and `value` would be unclear. It is clear in this context that `key` and `value` referred to login ID.